### PR TITLE
Design decision for storage API entry points on the bucket object

### PR DESCRIPTION
**Pull request by ayuishii**
Resolve TODO in [Bucket designation for each storage API](https://github.com/WICG/storage-buckets/blob/gh-pages/explainer.md#bucket-designations-for-each-storage-api) section.

### CODE REVIEW

File: explainer.md [hunk 1]
```diff
@@ -931,15 +931,22 @@ const inboxRegistration = await navigator.serviceWorker.register(
...
 The API for creating buckets and specifying / querying policies wouldn't change.
 
-TODO: Explain why this is worse than the main decision.
+This alternative was rejected because of worse ergonomics for web developers. 
+Web developers will have to specify bucket names in the options per storage API
+creating more opportunity for error by either forgetting to specify a name
+or typos in the bucket name.
 
-Developer ergonomics
+Having entry points to for each storage API on the bucket also makes replacing
```
Comment (pwnall) [inline explainer.md:939]: "to for" -> "to" ?

File: explainer.md [hunk 2]
```diff
@@ -931,15 +931,22 @@ const inboxRegistration = await navigator.serviceWorker.register(
 
 The API for creating buckets and specifying / querying policies wouldn't change.
 
-TODO: Explain why this is worse than the main decision.
+This alternative was rejected because of worse ergonomics for web developers. 
+Web developers will have to specify bucket names in the options per storage API
```
Comment (pwnall) [inline explainer.md:935]: "will have" -> "would have"?

File: explainer.md [hunk 3]
```diff
@@ -931,15 +931,22 @@ const inboxRegistration = await navigator.serviceWorker.register(
...
+creating more opportunity for error by either forgetting to specify a name
```
Comment (pwnall) [inline explainer.md:936]: opportunity -> opportunities
error -> errors

File: explainer.md [hunk 4]
```diff
@@ -931,15 +931,22 @@ const inboxRegistration = await navigator.serviceWorker.register(
...
+or typos in the bucket name.
```
Comment (pwnall) [inline explainer.md:937]: or typos -> or having typos?

(Parallel structure -- the other element in the list is "forgetting ...", which is a verb.)

File: explainer.md [hunk 5]
```diff
@@ -931,15 +931,22 @@ const inboxRegistration = await navigator.serviceWorker.register(
...
-* Harder to make mistakes - option dictionaries are prone to typos
-* Easy to replace the default bucket in JS -
-  `window.indexedDB = inboxBucket.indexedDB`
+```javascript
+const inboxBucket = await navigator.storageBuckets.openOrCreate("inbox", {
+  title: "Inbox",
+});
 
+// Replace default bucket with inboxBucket for IndexedDB. 
+window.indexedDB = inboxBucket.indexedDB;
```
Comment (pwnall) [inline explainer.md:948]: Please wire a couple more entry points and then have a `// ...` comment.

```js
window.caches = inboxBucket.caches;
navigator.storage.estimate = inboxBucket.estimate.bind(inboxBucket);
// ...
```

**Review by pwnall — CHANGES_REQUESTED** [review]
Thank you for describing how this alternative plays out!

**Comment by ayuishii** [comment]
Thanks for reviewing!

**Review by pwnall — APPROVED** [review]
LGTM, thank you for the revisions!

**Pull request merged**